metricbeat/helper/prometheus/textparse.go

```diff
@@ -404,21 +409,18 @@ func histogramMetricName(name string, s float64, qv string, lbls string, t *int6
 	case isCount(name):
 		u := uint64(s)
 		histogram.SampleCount = &u
-		name = name[:len(name)-6]
+		name = strings.TrimSuffix(name, suffixCount)
 	case isSum(name):
 		histogram.SampleSum = &s
-		name = name[:len(name)-4]
+		name = strings.TrimSuffix(name, suffixSum)
 	case isGaugeHistogram && isGCount(name):
 		u := uint64(s)
 		histogram.SampleCount = &u
-		name = name[:len(name)-7]
+		name = strings.TrimSuffix(name, suffixGCount)
 	case isGaugeHistogram && isGSum(name):
 		histogram.SampleSum = &s
-		name = name[:len(name)-5]
-	default:
-		if isBucket(name) {
-			name = name[:len(name)-7]
-		}
+		name = strings.TrimSuffix(name, suffixGSum)
+	case isBucket(name):
```

Comment:
Do we have any unit-tests that cover these changes? It's hard to understand if something might break here in any corner case or whatever? 

If not would you mind adding some tests? Otherwise we can exclude these changes from the patch if not needed for the fix. Improving code should always go along with adding tests ensuring that functionality is not broken :).

Reply:
I think that for the new functionality the added test is fine but what about the existing functionality that is touched here? How we can be sure that the code which is changed (improved) is not broken?
I'm talking mostly about the changes like the one at https://github.com/elastic/beats/pull/36537/files#diff-d66f15900f5676672ddf353458a9373cf8b1ab8e237c473191e4a64ae93cbba2L407-R416.

So the point here is that if we want to improve the code we should ensure that we don't break it. So to my mind we should either:

1) ensure that we have tests for these lines that still work
2) if we don't have tests then add them along with the corresponding code changes.
3) if we cannot add tests maybe it's safer to avoid any code changes that might be prone to errors

Could you verify the above cases please?

Reply:
We never had tests for those cases. Do you mean adding a test case for every time (histogram, histogram gauge etc)? I only changed it to `strings.TrimSuffix` because then we know what is happening instead of defining the lenght of the suffix and then subtracting like we were doing.

Reply:
> Do you mean adding a test case for every type (histogram, histogram gauge etc)?

Ideally yes since we touch these specific lines of code. Otherwise the improvement seems to be incomplete to me and potentially prone to errors :).
Maybe that's out of the scope of this PR but the generic idea is what I mention.

Reply:
I see. So should I try to add new tests for this PR or should we leave it to a different one?

Reply:
That would be nice since we are on this. Otherwise I doubt it will happen at any time soon.
Another option would be to exclude the specific changes/improvements that are not related to this specific fix and add them later in a follow-up along with tests. That would unblock this one and ensures the quality as well. It's up to you to decide.